### PR TITLE
8289710: Move Suspend/Resume classes out of os.hpp

### PR DESCRIPTION
**Pull request by iklam**
Please review this simple change that only renames a few classes and moved some code around. No functional changes.

The following classes are used only sparingly. They should be moved to a new header file share/runtime/suspend.hpp to minimize the size of os.hpp

- SuspendedThreadTaskContext
- SuspendedThreadTask
- SuspendResume

I didn't move the OS-specific implementation to a new file -- the POSIX implementation is currently inside [signals_posix.cpp](https://github.com/openjdk/jdk/blob/df063f7db18a40ea7325fe608b3206a6dff812c1/src/hotspot/os/posix/signals_posix.cpp#L1790) mixed with other signal handling code, so it doesn't seem a good idea to move out just the code for the above 3 classes.

The only other implementation is in os_windows.cpp. I could move the code to suspend_windows.cpp, but I don't feel very motivated unless someone insists.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8289710](https://bugs.openjdk.org/browse/JDK-8289710): Move Suspend/Resume classes out of os.hpp

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [8265ecb2](https://git.openjdk.org/jdk/pull/9371/files/8265ecb22e3d9687ce93b42ac4272776355e5570)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [8265ecb2](https://git.openjdk.org/jdk/pull/9371/files/8265ecb22e3d9687ce93b42ac4272776355e5570)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9371/head:pull/9371` \
`$ git checkout pull/9371`

Update a local copy of the PR: \
`$ git checkout pull/9371` \
`$ git pull https://git.openjdk.org/jdk pull/9371/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9371`

View PR using the GUI difftool: \
`$ git pr show -t 9371`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9371.diff">https://git.openjdk.org/jdk/pull/9371.diff</a>

</details>

### CODE REVIEW

**Review by dholmes-ora — COMMENTED** [review]
It is hard for me to see what is truly shared, what is posix-only and what is windows-only. Probably the existing placement is not very good when it comes to making those distinctions, but moving things into new header files just seem to highlight the problem.

I'm also concerned that this and other recent header files changes are breaking the convention that we generally have a foo.hpp and foo.cpp file. In some cases now if I see a declaration in a header file I have to go and search to find the right cpp file.

File: src/hotspot/share/runtime/suspend.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include "precompiled.hpp"
+#include "runtime/atomic.hpp"
+#include "runtime/suspend.hpp"
+
+void SuspendedThreadTask::run() {
+  internal_do_task();
+  _done = true;
+}
+
+#ifndef _WINDOWS
```
[the remark this answers is not in the document]
Comment (iklam) [reply src/hotspot/share/runtime/suspend.cpp]: The `SuspendResume` class is used only inside signals_posix.cpp (which requires the field `SuspendResume OsThread::sr` to be declared in osThread_{aix, bsd, linux}.hpp). I moved this class into the os/posix directory. I also renamed suspend.hpp to suspendedThreadTask.hpp.

File: src/hotspot/share/runtime/suspendedThreadTask.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "runtime/atomic.hpp"
```
Comment (dholmes-ora) [inline src/hotspot/share/runtime/suspendedThreadTask.cpp:26]: I don't think this is needed.

**Review by dholmes-ora — APPROVED** [review]
Looks good. Thanks.

**Comment by iklam** [comment]
Thanks @coleenp and @dholmes-ora for the review.
/integrate

**Pull request closed without merging**